jenkins/run-server-func-tests

```diff
@@ -75,11 +76,44 @@ function dump_journal {
     printf -- "\n--- journalctl dump ---\n\n"
 }
 
+function sc_disable {
+    # Run systemctl to disable the Pbench Server unit and clean up after any
+    # resulting failures attempting to do so.
+
+    # Send the command output to the bitbucket; capture stderr in a variable.
+    o=$(systemctl --user disable --now pbench-server 2>&1 >/dev/null)
+    r=${?}
+    if (( r != 0 )); then
+        # Ignore the expected error
+        if [[ ${o} != *"Unit file"*"does not exist"* ]]; then
+            echo "systemctl disable: ${o}"
+            return ${r}
+        fi
+    fi
+
+    # Even though we've disabled it above, the pbench-server service may still
+    # be present in systemctl's memory if it failed before or during being
+    # disabled.  A side-effect of the command below will purge this entry.
+    o=$(systemctl --user reset-failed pbench-server 2>&1 >/dev/null)
+    r=${?}
+    if (( r != 0 )); then
+        # Ignore the expected error
+        if [[ ${o} != *"Unit"*"not found"* && ${o} != *"Unit"*"not loaded"* ]]; then
+            echo "systemctl reset-failed: ${o}"
+            return ${r}
+        fi
+    fi
+
+    return ${r}
+}
+
 function cleanup {
-    if (( ${cleanup_flag} )); then
+    if (( cleanup_flag )); then
         # Remove the Pbench Server container and the dependencies pod which we
         # just created and ran; remove any dangling containers; and then remove
         # any dangling images.
+        echo "Disabling the pbench-server service..." >&2
+        sc_disable
         echo "Forcefully removing the Pbench Server container..." >&2
         podman rm --force --ignore ${PB_SERVER_CONTAINER_NAME}
```

Comment:
Given that control over the container is now encapsulated in the service file, I'm not sure this is a good idea. At best, it's a potential point of divergence and confusion.

Reply:
I suffer from a lack of faith in the ultimate effectiveness of the service file on the abort path, so I regard this as a belt-and-suspenders approach -- if `systemctl` is working as desired, then this shouldn't actually have any effect, positive or negative, but, if `systemctl` doesn't come through for us, this will remove the container (although, it won't prevent `systemctl` from trying to restart it...but hopefully we cover that elsewhere).